### PR TITLE
elboulangero/casync extract fixes

### PR DESCRIPTION
**Pull request by elboulangero**

### CODE REVIEW

File: src/update_handler.c [hunk 1]
```diff
@@ -275,6 +275,7 @@ static gboolean casync_extract_image(RaucImage *image, gchar *dest, GError **err
 
 	res = TRUE;
 out:
+	g_free(seed);
```
Comment (ejoerns) [inline src/update_handler.c]: Good catch! But should be solved in a different way as we've just introduced usage of glib automatic variable cleanup feature. Thus please use instead:
``` diff
--- a/src/update_handler.c
+++ b/src/update_handler.c
@@ -219,7 +219,7 @@ static gboolean casync_extract_image(RaucImage *image, gchar *dest, GError **err
        GError *ierror = NULL;
        gboolean res = FALSE;
        RaucSlot *seedslot = NULL;
-       gchar *seed = NULL;
+       g_autofree gchar *seed = NULL;
        gchar *store = NULL;
        gboolean seed_mounted = FALSE;
```

File: src/update_handler.c [hunk 2]
```diff
@@ -253,10 +253,11 @@ static gboolean casync_extract_image(RaucImage *image, gchar *dest, GError **err
 		seed = g_strdup(seedslot->device);
 	}
 
+extract:
+	/* Set store */
```
Comment (ejoerns) [inline src/update_handler.c]: Good catch, but please be a bit more verbose in your patch discription on why it fixes things. The actual thing that was wrong here is that `casync_extract()` would have been called without having initiallized the `store` variable.

**Comment by elboulangero** [comment]
Pull request updated, thanks for the feedback

**Comment by ejoerns** [comment]
Thanks for the update! Looks fine for me now. :+1:

**Pull request merged**